### PR TITLE
Update protocol documentation to fix typos

### PR DESCRIPTION
**Pull request by dvguruprasad**
Assumed user error codes start from 10000. Please correct if wrong.

### CODE REVIEW

File: doc/protocol.md [hunk 1]
```diff
@@ -88,7 +88,7 @@ like so:
 {
   "type":     "init",
   "msg_id":   1,
-  "node_id":  "n3",
+  "node_id":  "n1",
```
Comment (aphyr) [inline doc/protocol.md:91]: Why change this?

Comment (dvguruprasad) [reply doc/protocol.md:91]: The description below says "The node_id field indicates the ID of the node which is receiving this message: here, the node ID is "n1"".  Thought the one in the message is a typo?

Comment (aphyr) [reply doc/protocol.md:91]: Oh, yes, of course! That'll teach me to review patches first thing in the morning. ;-)

**Comment by aphyr** [comment]
Thank you!

**Pull request merged**